### DISCUSSION TITLE
How can I download a blob ?

### DISCUSSION THREAD

**Discussion by rambip** [rustwasm/gloo#169] [Q&A]
I know that you can use the web_sys function `Url::create_object_url_with_blob` but I am not able to do it.

I don't know how to convert from a gloo blob to a web_sys blob, and there is no way to create an url from a blob with gloo.

How could I do it ?

**Comment by ranile** [comment]
Can you not use any of the `read_as_*` functions?

See docs:
- [callbacks](https://docs.rs/gloo-file/0.2.0/gloo_file/callbacks/index.html)
- [futures](https://docs.rs/gloo-file/0.2.0/gloo_file/futures/index.html)

**Comment by rambip** [comment]
Ok nice !

But once I have my string-encoded url, how can I download it ?
Do I have to use web_sys ?

**Reply by ranile** [reply]
Same as you would in JavaScript. There's no way to directly "download" a blob. So yes, you'll need to use `web-sys`.

**Reply by mkroehnert** [reply]
@rambip I implemented file downloading for a single page app I wrote using the Seed framework.
You can take a look here: https://gitlab.com/mkroehnert/weightrs/-/blob/master/src/download.rs
It works for me on Firefox mobile and desktop although in one of them the filename is not set correctly.

The seed specific functions are typically just thin wrappers which you could reimplement yourself.
